graviti/utility/user.py

```diff
@@ -355,3 +357,145 @@ def update(self, __m: Any = (), **kwargs: _V) -> None:  # pylint: disable=argume
+        if not self._data.__contains__(key):
+            self._keys.append(key)
+
+        self._data.__setitem__(key, value)
+
+    def __delitem__(self, key: Union[int, str]) -> None:
+        if isinstance(key, int):
+            self._keys.__delitem__(key)
+            key = self._keys.__getitem__(key)
+        else:
+            self._keys.remove(key)
+
+        self._data.__delitem__(key)
+
+    def __contains__(self, key: Any) -> bool:
+        return self._data.__contains__(key)
+
+    def __iter__(self) -> Iterator[str]:
+        return self._keys.__iter__()
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, self.__class__):
+            return False
+
+        return self._data.__eq__(other._data) and self._keys.__eq__(other._keys)
+
+    def keys(self) -> Tuple[str, ...]:  # type: ignore[override]
+        """Get all element keys in order.
+
+        Returns:
+            A tuple containing all elements keys.
+
+        """
+        return tuple(self._keys)
+
+    def items(self) -> Tuple[Tuple[str, _V], ...]:  # type: ignore[override]
+        """Return a tuple of the ordered (key, value) pairs in dict.
+
+        Returns:
+            The ordered (key, value) pairs in dict.
+
+        """
+        return tuple((key, self._data.__getitem__(key)) for key in self._keys)
+
+    def values(self) -> Tuple[_V, ...]:  # type: ignore[override]
+        """Return a tuple of the values in dict.
+
+        Returns:
+            The values in dict.
+
+        """
+        return tuple(self._data.__getitem__(key) for key in self._keys)
+
+    def clear(self) -> None:
+        """Remove all items from the OrderedDict object."""
+        self._data.clear()
+        self._keys.clear()
+
+    @overload
+    def pop(self, key: str) -> _V:  # pylint: disable=arguments-differ
+        ...
+
+    @overload
+    def pop(self, key: str, default: Union[_V, _T] = ...) -> Union[_V, _T]:
+        ...
+
+    def pop(self, key: str, default: Any = __marker) -> Any:
+        """Remove specified item and return the corresponding value.
+
+        Arguments:
+            key: The key for dict.
+            default: The value to be returned if the key is not in the dict and it is given.
+
+        Returns:
+            Value to be removed from the OrderedDict object.
+
+        """
+        if self._data.__contains__(key):
+            self._keys.remove(key)
+
+        if default is self.__marker:
+            return self._data.pop(key)
+
+        return self._data.pop(key, default)
+
+    def popitem(self) -> Tuple[str, _V]:
+        """Remove and return a (key, value) pair as a tuple.
+
+        Pairs are returned in LIFO (last-in, first-out) order.
+
+        Returns:
+            A (key, value) pair as a tuple.
+
+        """
+        key = self._keys.pop()
+        return (key, self._data.pop(key))
+
+    def setdefault(self, key: str, default: _V = None) -> _V:  # type: ignore[assignment]
```

Comment:
int key?

Reply:
I think it is wired to support int key here. `__setitem__ should be used to set the value at a specific index.

Reply:
why it is wired?